### PR TITLE
feat: add basic support for example tag

### LINKED ISSUE #1201: Support `@example` tsdoc tag
The current implementation supports adding examples using a dedicated tag `@examples`.

But the TSDoc spec recommends using [`@example`](https://tsdoc.org/pages/tags/example/) instead.

While the TSDoc tag is mostly intended for documentation, and therefore does not always contain valid JSON data, the tag is still very much aligned with json-schema.

There are multiple levels of support which could be implemented:

### Basic

Just support raw JSON data

```ts
/**
 * @example
 *     { "foo": 42 }
 * @example
 *     "baz"
 */
```

Yields:
```json
{
    "examples": [
         { "foo": 42 },
        "baz"
    ]
}
```

An implementation of this is provided in #1200 .

### Support example title

The title would be ignored
```ts
/**
 * @example Title of this example
 *     { "foo": 42 }
 */
```

### Advanced

Would parse the markdown and extract the bits of code from the text. This is probably out of scope to be honest, but I'm including it here for the sake of completeness.

### PR DESCRIPTION
**Pull request by hmil**
fix #1201

# Version

Published prerelease version: `v1.0.1-next.1`

<details>
  <summary>Changelog</summary>

  :tada: This release contains work from a new contributor! :tada:
  
  Thank you, Hadrien Milano ([@hmil](https://github.com/hmil)), for all your work!
  
  #### 🚀 Enhancement
  
  - feat: add basic support for example tag [#1200](https://github.com/vega/ts-json-schema-generator/pull/1200) ([@hmil](https://github.com/hmil))
  
  #### 🐛 Bug Fix
  
  - fix: preserve empty `@description` [#1177](https://github.com/vega/ts-json-schema-generator/pull/1177) ([@Jason3S](https://github.com/Jason3S))
  
  #### 🔩 Dependency Updates
  
  - chore(deps-dev): bump @auto-it/conventional-commits from 10.34.2 to 10.36.5 [#1197](https://github.com/vega/ts-json-schema-generator/pull/1197) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.15.0 to 5.16.0 [#1185](https://github.com/vega/ts-json-schema-generator/pull/1185) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump node-fetch from 2.6.6 to 2.6.7 [#1195](https://github.com/vega/ts-json-schema-generator/pull/1195) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump minimist from 1.2.5 to 1.2.6 [#1196](https://github.com/vega/ts-json-schema-generator/pull/1196) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump json5 from 2.2.0 to 2.2.1 [#1198](https://github.com/vega/ts-json-schema-generator/pull/1198) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump prettier from 2.6.0 to 2.6.1 [#1199](https://github.com/vega/ts-json-schema-generator/pull/1199) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint from 8.11.0 to 8.12.0 [#1186](https://github.com/vega/ts-json-schema-generator/pull/1186) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @auto-it/first-time-contributor from 10.34.2 to 10.36.5 [#1187](https://github.com/vega/ts-json-schema-generator/pull/1187) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump auto from 10.34.2 to 10.36.5 [#1188](https://github.com/vega/ts-json-schema-generator/pull/1188) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump typescript from 4.6.2 to 4.6.3 [#1189](https://github.com/vega/ts-json-schema-generator/pull/1189) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump @types/json-schema from 7.0.10 to 7.0.11 [#1190](https://github.com/vega/ts-json-schema-generator/pull/1190) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/node from 17.0.21 to 17.0.23 [#1191](https://github.com/vega/ts-json-schema-generator/pull/1191) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.15.0 to 5.16.0 [#1192](https://github.com/vega/ts-json-schema-generator/pull/1192) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump vega from 5.22.0 to 5.22.1 [#1193](https://github.com/vega/ts-json-schema-generator/pull/1193) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump ajv from 8.10.0 to 8.11.0 [#1194](https://github.com/vega/ts-json-schema-generator/pull/1194) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  
  #### Authors: 3
  
  - [@dependabot[bot]](https://github.com/dependabot[bot])
  - Hadrien Milano ([@hmil](https://github.com/hmil))
  - Jason Dent ([@Jason3S](https://github.com/Jason3S))
</details>

### CODE REVIEW

File: src/AnnotationsReader/ExtendedAnnotationsReader.ts [hunk 1]
```diff
@@ -70,4 +72,38 @@ export class ExtendedAnnotationsReader extends BasicAnnotationsReader {
...
+        }
+
+        const jsDocTags: ts.JSDocTagInfo[] = symbol.getJsDocTags();
+        if (!jsDocTags || !jsDocTags.length) {
+            return undefined;
+        }
+
+        const examples: unknown[] = [];
+        for (const example of jsDocTags.filter((tag) => tag.name === "example")) {
+            console.log(JSON.stringify(example.text));
```
Comment (domoritz) [inline src/AnnotationsReader/ExtendedAnnotationsReader.ts]: remove log

File: test/valid-data/annotation-example/main.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * @example
+ * {
+ *     "nested": "hello"
+ * }
+ *
+ * @example Another example
```
Comment (domoritz) [inline test/valid-data/annotation-example/main.ts]:
```suggestion
 * @example An invalid example
```

File: test/valid-data/annotation-example/main.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * {
+ *     "nested": "world"
+ * }
+ * @example Last example
```
Comment (domoritz) [inline test/valid-data/annotation-example/main.ts]:
```suggestion
 * @example Another invalid example
```

**Review by domoritz — APPROVED** [review]

File: src/AnnotationsReader/ExtendedAnnotationsReader.ts (hunk 1 above)
Comment (hmil) [reply src/AnnotationsReader/ExtendedAnnotationsReader.ts]: whoops 😅

**Comment by domoritz** [comment]
Thank you.

**Pull request merged**